packages/lodestar/src/chain/factory/block/index.ts

```diff
@@ -30,25 +31,22 @@ export async function assembleBlock(
     stateRoot: ZERO_HASH,
     body: await assembleBody(config, db, eth1, stateContext.state, randaoReveal, graffiti),
   };
-
-  const preStateContext = await chain.regen.getBlockSlotState(head.blockRoot, slot - 1);
```

Comment:
Don't we still need this preStateContext?

Reply:
no, we don't need to in order to save 1 epoch transition possibly.
Current:
+ process head state up to slot
+ process head state up to (slot - 1) (which is redundant)
+ run `fastStateTransition` which is processing slot (epoch possibly) + process block
This PR changes to:
+ process head state up to slot
+ ~process head state to (slot - 1)~
+ ~run `fastStateTransition` which is processing slot (epoch possibly) + process block~ => process block only